### PR TITLE
[DS-171] RadioGroup.Radio 

### PR DESCRIPTION
**Pull request by ggdaltoso**
image

### CODE REVIEW

File: packages/yoga/src/RadioGroup/native/Radio/Radio.jsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+  ({
+    theme: {
+      yoga: {
+        components: { radioGroup },
+      },
+    },
+    checked,
+    disabled,
+  }) => `
+  positionL relative;
```
Comment (allyssonsantos) [inline packages/yoga/src/RadioGroup/native/Radio/Radio.jsx]:
```suggestion
  position: relative;
```

File: packages/yoga/src/RadioGroup/web/Button/RadioButton.jsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+import { HiddenInput } from '../../../shared';
+
+const Button = styled.label`
+  position: relative;
+  display: flex;
+  flex: 1;
+  align-items: center;
+  justify-content: center;
+
+  min-width: 126px;
```
Comment (allyssonsantos) [inline packages/yoga/src/RadioGroup/web/Button/RadioButton.jsx]: This value should be coming from theme, right?

File: packages/yoga/src/RadioGroup/native/Button/RadioButton.jsx [hunk 3]
```diff
@@ -20,17 +18,19 @@ const RadioMark = styled.View(
   align-items: center;
   justify-content: center;
   background-color: red;
-  background-color: ${radio.backgroundColor.enabled};
-  border-radius: ${radio.border.radius}px;
-  font-size: ${radio.font.size}px;
-  height: ${small ? radio.height.small : radio.height.normal}px;
-  padding-left: ${radio.padding.left}px;
-  padding-right: ${radio.padding.right}px;
+  background-color: ${radioGroup.button.backgroundColor};
```
Comment (victormath12) [inline packages/yoga/src/RadioGroup/native/Button/RadioButton.jsx]: What do you think to use the CSS Ordering here?

File: packages/yoga/src/RadioGroup/native/Radio/Radio.jsx (hunk 1 above)
Comment (victormath12) [inline packages/yoga/src/RadioGroup/native/Radio/Radio.jsx]:
```suggestion
  position: relative;
```

File: packages/yoga/src/RadioGroup/native/Radio/Radio.jsx [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+      },
+    },
+    checked,
+    disabled,
+  }) => `
+  height: ${radioGroup.radio.size * 0.5}px;
+  width: ${radioGroup.radio.size * 0.5}px;
+
+  background-color: ${radioGroup.radio.backgroundColor};
+
```
Comment (victormath12) [inline packages/yoga/src/RadioGroup/native/Radio/Radio.jsx]: You dont need to break line here bro.

File: packages/yoga/src/RadioGroup/native/Radio/Radio.jsx [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+  position: absolute;
+  height: ${radioGroup.radio.size + radioGroup.radio.size * 0.4 * 2}px;
+  width: ${radioGroup.radio.size + radioGroup.radio.size * 0.4 * 2}px;
+
+  border-width: ${radioGroup.radio.size * 0.4}px;
+  border-style: solid;
+  border-color: ${radioGroup.checked.hover.backgroundColor};
+  border-radius: ${radioGroup.radio.border.radius}px;
+
+  opacity: 0.75;
```
Comment (victormath12) [inline packages/yoga/src/RadioGroup/native/Radio/Radio.jsx]: This attribute should be in the theme, shouldn't it?

File: packages/yoga/src/RadioGroup/web/Radio/Radio.jsx [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+
+  ${({
+    theme: {
+      yoga: {
+        components: { radioGroup },
+      },
+    },
+  }) =>
+    `
+      width: ${radioGroup.radio.size}px;
```
Comment (alan-oliv) [inline packages/yoga/src/RadioGroup/web/Radio/Radio.jsx]: Can we descend the destructuring level to the "radio" level?

File: packages/yoga/src/RadioGroup/native/Radio/Radio.jsx [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+`,
+);
+
+/** The Radio is a type of selection control that allows the user to select a
+ * single option from a list.  */
+const RadioGroupRadio = ({ value, style, disabled, ...rest }) => {
```
Comment (victormath12) [inline packages/yoga/src/RadioGroup/native/Radio/Radio.jsx]: Do you really need this `style` prop? Can't u use it from `...rest` prop?

File: packages/yoga/src/RadioGroup/web/Button/RadioButton.jsx [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+    box-sizing: border-box;
+    background-color: ${radioGroup.button.backgroundColor};
+    border-radius: ${radioGroup.button.border.radius}px;
+
+    ${
+      checked
+        ? `
+            background-color: ${radioGroup.checked.backgroundColor};
+            box-shadow: 0 0 0  ${radioGroup.button.border.width}px ${radioGroup.checked.backgroundColor};
+            color: ${radioGroup.checked.textColor};
```
Comment (victormath12) [inline packages/yoga/src/RadioGroup/web/Button/RadioButton.jsx]:
```suggestion
            color: ${radioGroup.checked.font.color};
```

File: packages/yoga/src/RadioGroup/web/Radio/Radio.jsx [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+import { bool, oneOfType, string, number, shape } from 'prop-types';
+import styled from 'styled-components';
+import { hexToRgb } from '@gympass/yoga-common';
+
+import RadioGroupContext from '../../RadioGroupContext';
+import { HiddenInput } from '../../../shared';
+
+const Radio = styled.span`
+  position: relative;
+
```
Comment (victormath12) [inline packages/yoga/src/RadioGroup/web/Radio/Radio.jsx]: You don't need this breaking line, `display` and `position` is Layout Properties in the CSS Ordering

File: packages/yoga/src/RadioGroup/native/Radio/Radio.jsx (hunk 1 above)
Comment (alan-oliv) [inline packages/yoga/src/RadioGroup/native/Radio/Radio.jsx]: 👀

File: packages/yoga/src/RadioGroup/native/Radio/Radio.jsx [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+      onPress={() => {
+        onChange({ value: inputValue });
+      }}
+      onPressOut={() => {
+        togglePressing(false);
+      }}
+      disabled={disabled}
+      {...context}
+    >
+      <RadioMark checked={checked} disabled={disabled} {...rest} style={style}>
```
Comment (alan-oliv) [inline packages/yoga/src/RadioGroup/native/Radio/Radio.jsx]: Can we use the same object spread pattern that we use here: https://github.com/Gympass/yoga/pull/120/files#diff-95b56e2555cb4bdbfccb3e2f0079a13dR175 ?

**Pull request merged**